homeassistant/components/homekit/__init__.py

```diff
@@ -0,0 +1,122 @@
+
+HOMEKIT_FILE = '.homekit.state'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.All({
+        vol.Optional(CONF_PORT, default=51826): vol.Coerce(int),
+        vol.Optional(CONF_PIN_CODE, default='123-45-678'): cv.string,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Setup the homekit component."""
+    # pylint: disable=unused-variable
+    from .covers import Window  # noqa F401
+    # pylint: disable=unused-variable
+    from .sensors import TemperatureSensor # noqa F401
+    _LOGGER.debug("Begin setup homekit")
+
+    conf = config[DOMAIN]
+    port = conf.get(CONF_PORT)
+    pin = str.encode(conf.get(CONF_PIN_CODE))
+
+    homekit = Homekit(hass, port)
+    homekit.setup_bridge(pin)
+
+    hass.bus.async_listen_once(
+        EVENT_HOMEASSISTANT_START, homekit.start_driver)
+    return True
+
+
+def get_accessory(hass, state):
+    """Take state and return an accessory object if supported."""
+    if state.domain == 'sensor':
+        if state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == TEMP_CELSIUS:
+            _LOGGER.debug("Add \"%s\" as \"%s\"",
+                          state.entity_id, 'TemperatureSensor')
+            return TYPES['TemperatureSensor'](hass, state.entity_id,
+                                              state.name)
+
+    elif state.domain == 'cover':
+        # Only add covers that support set_cover_position
+        if state.attributes.get(ATTR_SUPPORTED_FEATURES) & 4:
+            _LOGGER.debug("Add \"%s\" as \"%s\"",
+                          state.entity_id, 'Window')
+            return TYPES['Window'](hass, state.entity_id, state.name)
+
+    return None
+
+
+class Homekit():
+    """Class to handle all actions between homekit and Home Assistant."""
+
+    def __init__(self, hass, port):
+        """Initialize a homekit object."""
+        self._hass = hass
+        self._port = port
+        self.bridge = None
+        self.driver = None
+
+    def setup_bridge(self, pin):
+        """Setup the bridge component to track all accessories."""
+        from .accessories import HomeBridge
+        self.bridge = HomeBridge(BRIDGE_NAME, pincode=pin)
+        self.bridge.set_accessory_info('homekit.bridge')
+
+    def add_accessory(self, acc):
+        """Add an accessory to the bridge."""
+        self.bridge.add_accessory(acc)
+
+    def start_driver(self, event):
+        """Start the accessory driver."""
+        from pyhap.accessory_driver import AccessoryDriver
+        _LOGGER.debug('Start driver')
+        self._hass.bus.async_listen_once(
+            EVENT_HOMEASSISTANT_STOP, self.stop_driver)
+
+        _LOGGER.debug('Start adding accessories.')
+        for state in self._hass.states.async_all():
+            acc = get_accessory(self._hass, state)
+            if acc is not None:
+                self.bridge.add_accessory(acc)
+
+        ip_address = get_local_ip()
+        # ip_address = yield from self._hass.async_add_job(get_local_ip)
```

Comment:
`ip_address = yield from self._hass.async_add_job(get_local_ip)` didn't work.

Reply:
Correct, because that's only valid inside async context. Inside a sync context, you can just call the method.

Reply:
You can remove the comment